src/mtype.c

```diff
@@ -5635,8 +5635,8 @@ Type *TypeFunction::semantic(Loc loc, Scope *sc)
                 }
                 else
                 {
-                    error(loc, "auto can only be used for template function parameters");
-                    errors = true;
+                    fparam->storageClass &= ~STCauto;
+                    fparam->storageClass |= STCrvref;
```

Comment:
Accepted yes, but not converted to an rvalue reference, since STCref is not present.